core/arch/arm/mm/core_mmu.c

```diff
@@ -289,6 +287,31 @@ static void add_phys_mem(struct tee_mmap_region *memory_map, size_t num_elems,
 	memory_map[n].size = mem->size;
 }
 
+static void add_va_space(struct tee_mmap_region *memory_map, size_t num_elems,
+			 unsigned int type, size_t size, size_t *last) {
+	size_t n = 0;
+
+	DMSG("type %d size 0x%08zx", type, size);
+	while (true) {
+		if (n >= (num_elems - 1)) {
+			EMSG("Out of entries (%zu) in memory_map", num_elems);
+			panic();
+		}
+		if (n == *last)
+			break;
+		if (type < memory_map[n].type)
```

Comment:
To keep entries sorted by type. There are huge comment in `add_phys_mem()` regarding this.